### PR TITLE
For #23433 - Align the provided top sites to the design spec

### LINKED ISSUE #23433: Align the provided top sites to the design spec
https://mozilla-hub.atlassian.net/browse/FNXV2-19282

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #23433 

- Change the margin between the favicon and title from 8dp to 6dp as outline in the designs
- Added subtitle below the top sites title that uses Secondary Text and 10sp font size.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/layout/top_site_item.xml [hunk 1]
```diff
@@ -41,6 +41,26 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:maxWidth="84dp"
+        android:layout_gravity="center_horizontal"
+        android:gravity="center"
+        android:textAlignment="center"
+        android:singleLine="true"
+        android:text="@string/top_sites_sponsored_label"
+        android:textColor="@color/fx_mobile_text_color_secondary"
+        android:textSize="10sp"
```
Comment (MozillaNoah) [inline app/src/main/res/layout/top_site_item.xml:58]: Is 10sp accessible?

File: app/src/main/res/layout/top_site_item.xml [hunk 2]
```diff
@@ -41,6 +41,26 @@
...
         app:layout_constraintTop_toBottomOf="@id/favicon_card"
-        tools:ignore="RtlCompat,SmallSp" />
+        tools:ignore="RtlCompat,SmallSp"
+        tools:text="Mozilla"/>
+
+    <TextView
+        android:id="@+id/top_site_subtitle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:maxWidth="84dp"
```
Comment (MozillaNoah) [inline app/src/main/res/layout/top_site_item.xml:51]: Is the max width a precaution so this doesn't grow beyond the width of a top site?

**Review by MozillaNoah — APPROVED** [review]
Looks good. Just have a couple small questions for clarification

File: app/src/main/res/layout/top_site_item.xml (hunk 1 above)
Comment (gabrielluong) [reply app/src/main/res/layout/top_site_item.xml:58]: I am not fully sure. It's part of the spec, but I will double check with @Verdi and adjust in a followup if needed.

File: app/src/main/res/layout/top_site_item.xml (hunk 2 above)
Comment (gabrielluong) [reply app/src/main/res/layout/top_site_item.xml:51]: I believe that is the intent.

**Pull request merged**